tutorials/hardhat-tutorial.md

```diff
@@ -0,0 +1,425 @@
+
+Next, a fixture is loaded - in this case, the "empty" stakingDAO fixture. Fixtures are easily interchangeable within your test cases. 
+
+```javascript
+// You can nest describe calls to create subsections.
+describe("Deployment", function () {
+    // `it` is another Mocha function. This is the one you use to define each
+    // of your tests. It receives the test name, and a callback function.
+    //
+    // If the callback function is async, Mocha will `await` it.
+    it("Should have correct target collator", async function () {
+      // We use loadFixture to setup our environment, and then assert that
+      // things went well
+      const { deployedDao, deployer } = await loadFixture(deployDaoFixture);
+
+      // `expect` receives a value and wraps it in an assertion object. These
+      // objects have a lot of utility methods to assert values.
+      expect(await deployedDao.target()).to.equal(targetCollator);
+});
+```
+
+Let's add another test case. When a StakingDAO is launched, it shouldn't have any funds. This test verifies that is indeed the case. Go ahead and add the following test case to your `Dao.js` file:
+
+```javascript
+it("The DAO should initially have 0 funds in it", async function () {
+	const { deployedDao, deployer } = await loadFixture(deployDaoFixture);
+	expect(await deployedDao.totalStake()).to.equal(0);
+});
+```
+
+### Function Reverts {: #function-reverts }
+
+To this point the two cases implemented have been simple but important. Now, you'll implement a more complex test case that differs in its architecture. In prior examples you've verified that a function returns an expected value. In this one, you'll be verifying that a function reverts. You'll also change the address of the caller to test an admin-only function. 
+
+In the [StakingDAO contract](https://github.com/PureStake/moonbeam-intro-course-resources/blob/main/delegation-dao-lesson-one/DelegationDAO.sol){target=_blank}, only admins are authorized to add new members to the DAO. One could write a test that checks to see if the admin is authorized to add new members but perhaps a more important test is to ensure that *non-admins* can't add new members. To run this test case under a different account, you're going to ask for another address when you call `ethers.getSigners()` and specify the caller in the assertion with `connect(otherAddress)`. Finally, after the function call to be tested you'll append `.to.be.reverted;` to indicate that the test case is successful if the function reverts. And if it doesn't revert it's a failed test! 
+
+```javascript
+it("Non-admins should not be able to grant membership", async function () {
```

Comment:
test descriptions should read like a sentence, so "it should not allow non-admints to grant membership"
```suggestion
it("should not allow non-admins to grant membership", async function () {
```